snap/validate.go

```diff
@@ -509,6 +546,9 @@ func ValidateLayout(layout *Layout, constraints []LayoutConstraint) error {
 	if layout.Bind != "" {
 		nused++
 	}
+	if layout.BindFile != "" {
+		nused++
+	}
 	if layout.Type != "" {
 		nused++
 	}
```

Comment:
I see that previous PRs include the nused validation, but I don't see a negative test where nused > 1 in validate_test.go or info_snap_yaml_test.go? Am I missing something? If not, can you add a test?

Reply:
It's there, in TestValidateLayout. I added a few more cases so that it's more numerous in the diff.

Reply:
Ah, mocking the data structure instead of seeing the yaml made it harder to see. Thanks!